src/main/kotlin/app/CodeLongevity.kt

```diff
@@ -0,0 +1,266 @@
+// Copyright 2017 Sourcerer Inc. All Rights Reserved.
+// Author: Alexander Surkov (alex@sourcerer.io)
+
+package app
+
+import org.eclipse.jgit.diff.DiffFormatter
+import org.eclipse.jgit.diff.DiffEntry
+import org.eclipse.jgit.diff.RawText
+import org.eclipse.jgit.internal.storage.file.FileRepository
+import org.eclipse.jgit.lib.ObjectId
+import org.eclipse.jgit.lib.ObjectLoader
+import org.eclipse.jgit.lib.Repository
+import org.eclipse.jgit.revwalk.RevCommit
+import org.eclipse.jgit.revwalk.RevWalk
+import org.eclipse.jgit.treewalk.TreeWalk
+import org.eclipse.jgit.util.io.DisabledOutputStream
+
+import java.text.SimpleDateFormat
+import java.util.Date
+
+/**
+ * Represents a code line in a file revision.
+ */
+class RevCommitLine(val commit: RevCommit, val file: String, val line: Int)
+
+/**
+ * Represents a code line in repo's history.
+ *
+ * TODO(Alex): the text arg is solely for testing proposes (remove it)
+ */
+class CodeLine(val from: RevCommitLine, val to: RevCommitLine, val text: String) {
+
+    // TODO(alex): oldId and newId may be computed as a hash built from commit,
+    // file name and line number, if we are going to send the data outside a
+    // local machine.
+
+    /**
+     * Id of the code line in a revision when the line was added. Used to
+     * identify a line and update its lifetime computed at the previous
+     * iteration.
+     */
+    val oldId: String = ""
+
+    /**
+     * Id of the code line in a revision, where the line was deleted, or a head
+     * revision, if the line is alive.
+     */
+    val newId: String = ""
+
+    /**
+     * The code line's age in seconds.
+     */
+    val age = to.commit.getCommitTime() - from.commit.getCommitTime()
+
+    /**
+     * A pretty print of a code line; debugging.
+     */
+    fun printme() {
+        val df = SimpleDateFormat("yyyy-MM-dd HH:mm z")
+        val fd = df.format(Date(from.commit.getCommitTime().toLong() * 1000))
+        val td = df.format(Date(to.commit.getCommitTime().toLong() * 1000))
+        val fc = "${from.commit.getName()} '${from.commit.getShortMessage()}'"
+        val tc = "${to.commit.getName()} '${to.commit.getShortMessage()}'"
+        println("Line '$text' - '${from.file}:${from.line}' added in $fc $fd")
+        println("  last known as '${to.file}:${to.line}' in $tc $td")
+    }
+}
+
+/**
+ * Used to compute age of code lines in the repo.
+ */
+class CodeLongevity(repoPath: String, tailRev: String) {
+    val repo = FileRepository(repoPath)
+    val head: RevCommit = RevWalk(repo).parseCommit(repo.resolve("HEAD"));
```

Comment:
do you have on top of your head how to get a master branch? getOriginHead apparently doesn't work for local repos.